### PR TITLE
Plugins: Skips existence of module.js for renderer plugins

### LINKED ISSUE #19130: Plugins: Warning is logged when Grafana starts up
**What happened**:
When starting Grafana the following warning message is logged:
```
WARN[09-16|10:13:27] Plugin missing module.js                 logger=plugins name="Grafana Image Renderer" warning="Missing module.js, If you loaded this plugin from git, make sure to compile it." path=/home/marcus/go/src/github.com/grafana/grafana/data/
plugins/grafana-image-renderer/module.js
```

**What you expected to happen**:
No warning logged since this is a renderer plugin which doesn't have any frontend parts.

**How to reproduce it (as minimally and precisely as possible)**:
1. Install grafana-image-renderer plugin
2. Restart Grafana
3. Inspect log

**Anything else we need to know?**:

**Environment**:
- Grafana version: master (7ace80c71cce5d3e0065590e3c9d2aee1363d3f9)

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #19130

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/plugins/plugins.go [hunk 1]
```diff
@@ -217,7 +217,7 @@ func (scanner *PluginScanner) loadPluginJson(pluginJsonFilePath string) error {
 	loader = reflect.New(reflect.TypeOf(pluginGoType)).Interface().(PluginLoader)
 
 	// External plugins need a module.js file for SystemJS to load
-	if !strings.HasPrefix(pluginJsonFilePath, setting.StaticRootPath) {
+	if !strings.HasPrefix(pluginJsonFilePath, setting.StaticRootPath) && pluginCommon.Type != "renderer" {
```
Comment (torkelo) [inline pkg/plugins/plugins.go]: can we extract this to && !isBackendOnly(pluginCommon.Type)

function isBackendOnly(string type) ... 

Should be clearer and we will very likely add more plugins that only have backend parts

**Review by torkelo — APPROVED** [review]

File: pkg/plugins/plugins.go (hunk 1 above)
Comment (hugohaggmark) [reply pkg/plugins/plugins.go]: Good suggestion, I will take a look again.

**Comment by hugohaggmark** [comment]
Are the changes ok "go-wise" @marefr @torkelo?

**Review by torkelo — APPROVED** [review]

**Comment by torkelo** [comment]
Yes!

**Comment by marefr** [comment]
Great! Nitpicking, I would have made the `IsBackendOnlyPlugin` begin with lowercase like `isBackendOnlyPlugin` to make it non-exportable from the package/internal - no other package would be able to call the method.

**Comment by hugohaggmark** [comment]
> Great! Nitpicking, I would have made the `IsBackendOnlyPlugin` begin with lowercase like `isBackendOnlyPlugin` to make it non-exportable from the package/internal - no other package would be able to call the method.

Thanks for that feedback, didn't know about that convention.

**Pull request merged**